### DISCUSSION TITLE
Type 'Observable<void>' is not assignable to type 'Observable<Action>'.

### DISCUSSION THREAD

**Discussion by giovannidias1** [ngrx/platform#3461] [Q&A] [answered]
I can't figure out what's wrong it looks like everything is right as in the tutorial

```
  loadCategories$ = createEffect(() =>
    this.actions$.pipe(
      ofType('[Question Component] load categories'),
      switchMap(() =>
        this.questionService.getCategories().pipe(
          map(data => console.log(data)),
          catchError(() => EMPTY)
        )
      )
    )
  );
```

> [webpack-dev-server] ERROR
> src/app/root-store/question-store/question.effects.ts:12:5 - error TS2322: Type 'Observable<void>' is not assignable to type 'EffectResult<Action>'.
>   Type 'Observable<void>' is not assignable to type 'Observable<Action>'.
>     Type 'void' is not assignable to type 'Action'.
> 
>  12     this.actions$.pipe(
>         ~~~~~~~~~~~~~~~~~~~
>  13       ofType('[Question Component] load categories'),
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ... 
>  19       )
>     ~~~~~~~
>  20     )
>     ~~~~~
> 
>   node_modules/@ngrx/effects/src/effect_creator.d.ts:42:161
>     42 export declare function createEffect<C extends EffectConfig, DT extends DispatchType<C>, OT extends ObservableType<DT, OT>, R extends EffectResult<OT>>(source: () => R & ConditionallyDisallowActionCreator<DT, R>, config?: Partial<C>): R & CreateEffectMetadata;
>                                                                                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     The expected type comes from the return type of this signature.

**Comment by Harpush** [comment, marked as answer]
I am pretty sure `map(data => console.log(data))` is supposed to use `tap` as you are mapping to console log which is 99% wrong.
